tools/16SDetector/hmmer-3.0/easel/miniapps/esl-compalign.man
.TH "esl-compalign" 1 "@EASEL_DATE@" "Easel @PACKAGE_VERSION@" "Easel miniapps"

.SH NAME
.TP 
esl-compalign - compare two multiple sequence alignments



.SH SYNOPSIS
.B esl-compalign
.I [options]
.I trusted_file
.I test_file



.SH DESCRIPTION

.I esl-compalign
evaluates the accuracy of a predicted multiple sequence alignment with
respect to a trusted alignment of the same sequences. 

The 
.I trusted_file 
and
.I test_file
must contain the same number of alignments. Each predicted alignment in the 
.I test_file 
will be compared against a single trusted alignment from the
.I trusted_file.
The first alignments in each file correspond to each other and will be
compared, the second alignment in each file correspond to each other
and will be compared, and so on.  Each corresponding pair of
alignments must contain the same sequences (i.e. if they were
unaligned they would be identical) in the same order in both
files. Further, both alignment files must be in Stockholm format and
contain 'reference' annotation, which appears as "#=GC RF" per-column
markup for each alignment. The number of nongap (non '.' characters)
in the reference (RF) annotation must be identical between all
corresponding alignments in the two files.

.I esl-compalign
reads an alignment from each file, and compares them based on
their 'reference' annotation.  The number of correctly predicted
residues for each sequence is computed as follows. A residue that is
in the Nth nongap RF column in the trusted alignment must also appear
in the Nth nongap RF column in the predicted alignment to be counted
as 'correct', otherwise it is 'incorrect'. A residue that appears in a
gap RF column in the trusted alignment between nongap RF columns N and
N+1 must also appear in a nongap RF column in the predicted alignment
between nongap RF columns N and N+1 to be counted as 'correct',
otherwise it is incorrect.

The default output of
.B esl-compalign
lists each sequence and the number of correctly and incorrectly
predicted residues for that sequence. These counts are broken down
into counts for residues in the predicted alignments that occur
in 'match' columns and 'insert' columns. A 'match' column is one for
which the RF annotation does not contain a gap. An 'insert' column is
one for which the RF annotation does contain a gap.

.SH OPTIONS

.TP
.B -h
Print brief help; includes version number and summary of
all options.

.TP
.B -c
Print per-column statistics instead of per-sequence statistics.

.TP
.B -p 
Print statistics on accuracy versus posterior probability values. The 
.I test_file
must be annotated with posterior probabilities (#=GR PP) for this
option to work.

.SH EXPERT OPTIONS

.TP
.BI --p-mask " <f>"
This option may only be used in combination with the 
.B -p
option. Read a "mask" from file 
.I <f>.
The mask file must consist of a single line, of only '0' and '1'
characters. There must be exactly RFLEN characters where RFLEN is the
number of nongap characters in the RF annotation of all alignments in
both 
.I trusted_file
and
.I test_file.
Positions of the mask that are '1' characters indicate that the
corresponding nongap RF position is included by the mask. The
posterior probability accuracy statistics for match columns will only
pertain to positions that are included by the mask, those that are
excluded will be ignored from the accuracy calculation.

.BI --c2dfile " <f>"
Save a 'draw file' to file 
.I <f>
which can be read into the 
.B esl-ssdraw
miniapp. This draw file will define two postscript pages for 
.B esl-ssdraw.
The first page will depict the frequency of errors per match position and
frequency of gaps per match position, indicated by magenta and yellow,
respectively. The darker magenta, the more errors and the darker
yellow, the more gaps. The second page will depict the frequency of
errors in insert positions in shades of magenta, the darker the
magenta the more errors in inserts after each position. See
.B esl-ssdraw
documentation for more information on these diagrams. 

.TP
.B --amino
Assert that 
.I trusted_file
and 
.I test_file
contain protein sequences. 

.TP 
.B --dna
Assert that 
.I trusted_file
and 
.I test_file
contain DNA sequences. 

.TP 
.B --rna
Assert that the 
.I trusted_file
and 
.I test_file
contain RNA sequences. 

.SH AUTHOR

Easel and its documentation are @EASEL_COPYRIGHT@.
@EASEL_LICENSE@.
See COPYING in the source code distribution for more details.
The Easel home page is: @EASEL_URL@


tools/16SDetector/hmmer-3.0/easel/miniapps/esl-compalign.itest.pl
#! /usr/bin/perl

# Integrated test of the esl-compalign miniapp.
#
# Usage:     ./esl-compalign.itest.pl <esl-compalign binary> <tmpfile prefix>
# Example:   ./esl-compalign.itest.pl ./esl-compalign        foo
#
# EPN, Tue Feb  2 13:19:44 2010

$eslcompalign= shift;
$tmppfx      = shift;

if (! -x "$eslcompalign") { die "FAIL: didn't find esl-compalign binary $eslcompalign"; }

open(ALIFILE, ">$tmppfx.1") || die "FAIL: couldn't open $tmppfx.1 for writing alifile";
print ALIFILE << "EOF";
# STOCKHOLM 1.0
#=GF AU Infernal 0.1

simpex-17          --AGA-CUUCGG---GCUCG-UAACAG
#=GR simpex-17 PP  ..885.*****9...9****.******
simpex-39          aaAAUACGUCGGCUG-AAUCACCAGUA
#=GR simpex-39 PP  ***************.67776899999
simpex-82          --ACGUUUUG-GAACGGG-U-CCA-CC
#=GR simpex-82 PP  ..****9998.88**888.5.898.9*
#=GC SS_cons       ..::<<<____>->>-<<-<.___>>>
#=GC RF            ..AAgaCUUCGGAucgggCg.AcAccc
//
EOF
close ALIFILE;

open(ALIFILE, ">$tmppfx.2") || die "FAIL: couldn't open $tmppfx.2 for writing alifile";
print ALIFILE << "EOF";
# STOCKHOLM 1.0
#=GF AU Infernal 0.1

simpex-17          ---AGACUUCGGG---CUCGUAACAG
#=GR simpex-17 PP  ...69*****775...4466777888
simpex-39          aaAAUACGUCGGCUGAAUCACCAGUA
#=GR simpex-39 PP  **************************
simpex-82          --ACGUUUUG-GAACGGGUC-C-ACC
#=GR simpex-82 PP  ..99998886.777755544.2.358
#=GC SS_cons       ..::::::::::::::::::::::::
#=GC RF            ..AAgACUUCGGAucggGCaAcAuUc
//
EOF
close ALIFILE;

open(MASKFILE, ">$tmppfx.mask") || die "FAIL: couldn't open $tmppfx.mask for writing alifile";
print MASKFILE << "EOF";
110111011011101110011101
EOF
close MASKFILE;

$output = `$eslcompalign -h`;
if ($? != 0)                                         { die "FAIL: esl-compalign failed unexpectedly"; }
if ($output !~ /Usage: esl-compalign/)                { die "FAIL: help output not right"; }

$output = `$eslcompalign --rna $tmppfx.1 $tmppfx.2 2>&1`;
if ($? != 0)                                                                                                                                { die "FAIL: esl-compalign failed unexpectedly";}
if ($output !~ /simpex-17      20        16 \/       20  \(0.80\d+\)         0 \/        0  \(0.000\)        16 \/       20  \(0.80\d+\)/)      { die "FAIL: alignments compared incorrectly"; }
if ($output !~ /\# \*all\*           -        53 \/       64  \(0.82\d+\)         2 \/        3  \(0.66\d+\)        55 \/       67  \(0.82\d+\)/) { die "FAIL: alignments compared incorrectly"; }

$output = `$eslcompalign -c $tmppfx.1 $tmppfx.2 2>&1`;
if ($? != 0)                                                                                    { die "FAIL: esl-compalign failed unexpectedly"; }
if ($output !~ /18     1 \/    3  \(0.33\d+\)     0 \/    1  \(0.00\d+\)     1 \/    4  \(0.25\d+\)/) { die "FAIL: alignments compared incorrectly."; }

$output = `$eslcompalign -p $tmppfx.1 $tmppfx.2 2>&1`;
if ($? != 0)                                                                                    { die "FAIL: esl-compalign failed unexpectedly"; }
if ($output !~ /\*        23 \/       29 \(0.79\d+\)         2 \/        2 \(1.0\d+\)/)        { die "FAIL: alignments compared incorrectly."; }

$output = `$eslcompalign --p-mask $tmppfx.mask -p $tmppfx.1 $tmppfx.2 2>&1`;
if ($? != 0)                                                                                    { die "FAIL: esl-compalign failed unexpectedly"; }
if ($output !~ /\*        17 \/       21 \(0.80\d+\)         2 \/        2 \(1.00\d+\)/)        { die "FAIL: alignments compared incorrectly."; }

system("$eslcompalign -c --c2dfile $tmppfx.dfile $tmppfx.1 $tmppfx.2 > /dev/null");
if ($? != 0)                                      { die "FAIL: esl-compalign failed unexpectedly"; }
$output = `cat $tmppfx.dfile`;
if ($output !~ /^0.000 0.33\d+ 0.000 0.000/)        { die "FAIL: alignments compared incorrectly."; }

print "ok\n"; 
unlink "$tmppfx.1";
unlink "$tmppfx.2";
unlink "$tmppfx.dfile";
unlink "$tmppfx.mask";
exit 0;
